### PR TITLE
Modules scale

### PR DESCRIPTION
**Pull request by dvdherron**
## Description
Upgrade the `Scale` module to use Sass modules

### CODE REVIEW

**Comment by dvdherron** [comment]
[I want Accoutrement on Sass Modules](https://trello.com/c/9tyINy2o/23-i-want-accoutrement-on-sass-modules)

**Comment by dvdherron** [comment]
no rush @mirisuzanne @stacyk but I'm stuck trying to figure out an issue with a value returning `null` from the `size` function in `./sass/scale/size` 

If you try to run the second test in ./test/scale/size ("Get size from unit adjustment"), it errors with this message: 
```
Error: $number: null is not a number.
    ╷
120 │   $from-unit: math.unit($length);
    │               ^^^^^^^^^^^^^^^^^^
    ╵
  sass/scale/_units.scss 120:15                          convert-units()
  sass/tokens/_parse.scss 185:13                         do()
  sass/tokens/_parse.scss 334:15                         compile-token()
  sass/tokens/_api.scss 295:16                           get()
  sass/scale/_size.scss 46:10                            size()
  test/scale/_size.scss 21:29                            @content
  node_modules/sass-true/sass/test/_test.scss 50:5       @content
  node_modules/sass-true/sass/test/_test.scss 24:3       test()
  node_modules/sass-true/sass/test/_test.scss 49:3       it()
  test/scale/_size.scss 20:5                             @content
```

The ` $length` value comes from the `size` function (called from the `convert-units` function, line #119 in `sass/scale/units`)

If I check what `$length` is just before line #119 with `@debug $length`, I get `20px`. 
If I make the same `@debug` statement immediately after line #119, I get `null`.

Any idea what could be the cause of this?

File: sass/scale/_size.scss [hunk 1]
```diff
@@ -43,8 +43,19 @@
...
   }
 
-  $size: api.get($source, $size, $do);
-  @return if((meta.type-of($size) == 'string'), string.unquote($size), $size);
+  $return: $size;
+
+  @if (meta.type-of($size) == 'number') {
+    // apply adjustments
+    @if $do {
+      $return: tokens.compile($source, $size $do);
```
Comment (dvdherron) [inline sass/scale/_size.scss:51]: @mirisuzanne should this be `api.compile`?

File: test/scale/_size.scss [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+    @include it('Allow viewport units without warning') {
+      @include assert-equal(size.size('percent'), 20%);
+    }
+
+    @include it('Allow simple calc values') {
+      @include assert-equal(size.size('calc-simple'), calc(1em + 2vw));
+    }
+
+/*     @include it('Allow compiled calc values') {
```
Comment (dvdherron) [inline test/scale/_size.scss:84]: There seems to be something in the test-data map that is being converted incorrectly? I get this error on the last three commented out tests in `size`

```
Error: ("times": 3) isn't a valid CSS value.
   ╷
40 │   @return if(($type == 'string'), $value, '#{$value}');
   │                                              ^^^^^^
   ╵
  sass/utils/_string.scss 40:46                          force()
  sass/utils/_string.scss 115:11                         replace()
  sass/utils/_string.scss 175:16                         interpolate()
  sass/tokens/_parse.scss 185:13                         do()
  sass/tokens/_parse.scss 334:15                         compile-token()
  sass/tokens/_api.scss 295:16                           get()
  sass/scale/_size.scss 55:14                            size()
```
They all appear to be using this section of the test-data map:

  ```
'calc': (
    '%value': 'calc(%s + (math.div(%s, %s)))',
    '%s': ('#root' ('times': 3), '#test-value' ('convert-units': 'vw'), 3),
  ),

Is there a typo in the `calc` map here that's causing these tests to fail @mirisuzanne ?
```

**Review by mirisuzanne — COMMENTED** [review]
a few small notes, but generally looking good!

File: sass/scale/_index.scss [hunk 3]
```diff
@@ -7,42 +7,30 @@
...
 /// ```
 /// @group scale
 
-
 /// ## Sizes & Ratios
 /// Establish your palette of ratios (modular scales) and sizes:
 ///
 /// ```scss
-/// $ratios: (
+/// tools.$ratios: (
```
[the remark this answers is not in the document]
Comment (dvdherron) [reply sass/scale/_index.scss]: aha, yes, forgot that distinction. removed these.

File: sass/scale/_index.scss [hunk 4]
```diff
@@ -7,42 +7,30 @@
...
 ///   'my-ratio': 1.25,
 /// );
 ///
-/// $sizes: (
+/// tools.$sizes: (
```
[the remark this answers is not in the document]
Comment (dvdherron) [reply sass/scale/_index.scss]: removed

File: sass/scale/_index.scss [hunk 5]
```diff
@@ -70,7 +58,7 @@
 /// ```scss
 /// .example {
 ///   font-size('text');
-///   width: size('page', 'px');
+///   width: tools.size('page', 'px');
```
[the remark this answers is not in the document]
Comment (dvdherron) [reply sass/scale/_index.scss]: removed

**Review by mirisuzanne — APPROVED** [review]

**Review by mirisuzanne — APPROVED** [review]

**Pull request merged**